backend/ddang/src/test/java/com/ddang/ddang/chat/infrastructure/persistence/QuerydslMessageRepositoryTest.java

```diff
@@ -0,0 +1,178 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayNameGeneration;
+import org.junit.jupiter.api.DisplayNameGenerator;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.context.annotation.Import;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@DataJpaTest
+@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+@SuppressWarnings("NonAscillCharacters")
+@Import({JpaConfiguration.class, QuerydslConfiguration.class})
+class QuerydslMessageRepositoryTest {
+
+    @PersistenceContext
+    EntityManager em;
+
+    @Autowired
+    JpaUserRepository userRepository;
+
+    @Autowired
+    JpaAuctionRepository auctionRepository;
+
+    @Autowired
+    JpaRegionRepository regionRepository;
+
+    @Autowired
+    JpaCategoryRepository categoryRepository;
+
+    @Autowired
+    JpaChatRoomRepository chatRoomRepository;
+
+    @Autowired
+    JpaMessageRepository messageRepository;
+
+    QuerydslMessageRepository querydslMessageRepository;
+
+    @BeforeEach
+    void setUp(@Autowired final JPAQueryFactory queryFactory) {
+        querydslMessageRepository = new QuerydslMessageRepositoryImpl(queryFactory);
+    }
+
+    @Test
+    void 마지막으로_읽은_메시지_이후에_추가된_메시지를_조회한다() {
+        // given
+        final User seller = User.builder()
+                                .name("회원")
+                                .profileImage("profile.png")
+                                .reliability(4.7d)
+                                .oauthId("78923")
+                                .build();
+        final User buyer = User.builder()
+                               .name("구매자")
+                               .profileImage("profile.png")
+                               .reliability(4.7d)
+                               .oauthId("12345")
+                               .build();
+        final Auction auction = Auction.builder()
+                                       .title("title")
+                                       .build();
+
+        userRepository.save(seller);
+        userRepository.save(buyer);
+        auctionRepository.save(auction);
+
+        final ChatRoom chatRoom = new ChatRoom(auction, buyer);
+        chatRoomRepository.save(chatRoom);
+
+        final int messagesCount = 10;
+        for (int count = 0; count < messagesCount; count++) {
+            final Message message = Message.builder()
+                                           .chatRoom(chatRoom)
+                                           .writer(seller)
+                                           .receiver(buyer)
+                                           .contents("안녕하세요")
+                                           .build();
+
+            messageRepository.save(message);
+        }
```

Comment:
작성해주신 테스트만으로는 `user가 receiver와 일치한 경우도 메시지를 가져올 수 있도록 변경` 이 제대로 잘 이루어졌는지 확인이 어려울 것 같습니다.

```suggestion
        final int messagesCount = 5;
        for (int count = 0; count < messagesCount; count++) {
            final Message message1 = Message.builder()
                                           .chatRoom(chatRoom)
                                           .writer(seller)
                                           .receiver(buyer)
                                           .contents("안녕하세요")
                                           .build();
           final Message message2 = Message.builder()
                                           .chatRoom(chatRoom)
                                           .writer(buyer)
                                           .receiver(seller)
                                           .contents("반갑습니다")
                                           .build();

            messageRepository.save(message1);
            messageRepository.save(message2);
        }
```

Reply:
이 부분도 마찬가지로 message를 조회하는 유저 id가 receiver인지, writer인지로 구분해주었습니다